### PR TITLE
Fix cross-comp CI pipeline (ARM & MIPS32) to build against target branch instead of main

### PR DESCRIPTION
**Pull request by ewertons**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/main/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `main` branch. 
  - [x] I have merged the latest `main` branch prior to submission and re-merged as needed after I took any feedback.
  - [x] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

The ARM and the MIPS32 cross-compilation jobs in .vsts-ci.yml will always target the main branch, ignoring the branch a given PR is based on and letting potential issues sip through. 

# Description of the solution
 
This change fixes the ARM and MIPS32 jobs so they run against whatever branch the pipeline is running against.

### CODE REVIEW

File: samples/dockerbuilds/ARM/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 FROM ubuntu:latest
 ARG DEBIAN_FRONTEND=noninteractive
+ARG TARGET_BRANCH=main
```
Comment (vaavva) [inline samples/dockerbuilds/ARM/Dockerfile:3]: Should this be the source branch?

**Review by vaavva — APPROVED** [review]

File: samples/dockerbuilds/ARM/Dockerfile (hunk 1 above)
Comment (CIPop) [reply samples/dockerbuilds/ARM/Dockerfile:3]: Looks like --build-arg will override: https://docs.docker.com/build/guide/build-args/

**Review by CIPop — APPROVED** [review]
LGTM
Because of a known breaking issue the build is _expected_ to fail with an include location failure. 

The build should still show the docker images being built correctly, out of the PR branch (instead of `main`). If this is true, @ewertons and I agreed to force-push the CI change.

File: samples/dockerbuilds/ARM/Dockerfile (hunk 1 above)
Comment (ewertons) [reply samples/dockerbuilds/ARM/Dockerfile:3]: Yes. This is the default value, "main". Its value get set by the pipeline job in .vsts-ci.yml

**Pull request merged**